tm-parsers/js/js.tm

```diff
@@ -1578,12 +1586,16 @@ TypeQuery -> TypeQuery :
     'typeof' TypeQueryExpression ;
 
 # 2.9
-ImportType -> ImportType :
-    'typeof'? 'import' '(' Type ')' ('.' TypeQueryExpression)? .noLineBreak TypeArguments? %prec resolveShift ;
+ImportType -> TsImportType:
+     ImportTypeStart .noLineBreak TypeArguments? %prec resolveShift ;
+
+ImportTypeStart -> TsImportTypeStart :
+  ('typeof'->TsTypeOf)? 'import' '(' Type ')'
+   | ImportTypeStart '.' IdentifierReference<~Yield, ~Await> -> TsImportTypeStart;
```

Comment:
please put semicolon on a separate line and align the two rules vertically at 4 spaces.